### PR TITLE
(#97) term count placeholder:

### LINKED ISSUE #97: Enabler: Term Count Placeholder for Intro Text
# TBD

Just the front-end portion of putting the term count placeholder in place in the intro text - passed in through the initialize (endpoint for count already exists in API).

Support transforming:

```
"The NCI Dictionary of Cancer Terms features ${term_count} terms related to cancer and medicine.

We offer a widget that you can add to your website to let users look up cancer-related terms. Get NCI’s Dictionary of Cancer Terms Widget."

```
To:

```
The NCI Dictionary of Cancer Terms features 8,531 terms related to cancer and medicine.

We offer a widget that you can add to your website to let users look up cancer-related terms. Get NCI’s Dictionary of Cancer Terms Widget.

```
Examples:

*Examples:*
* *Initial Setup of Dev Tier*
* *Create Application Framework*
* *Define Unit Test Approach*
* *Research UX Patterns for Search*

> **ESTIMATE** 2

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* References: *URL to web site with examples*
* Web Page: *Here goes a URL to an existing web page*

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Sub-Tasks

- [x] Create a method to replace any number of tokens in the given string and return modified string
- [x] Util to convert given number to thousands format

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…
* More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by kate-mashkina**
close #97 

- added new endpoint
- new setupProxy method
- mock data
- added an util to format any number to be displayed
with comma separated thousands
- added unit tests
- added token parser util

### CODE REVIEW

File: src/utils/replaceTokens.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        } else if (token !== null && Context[token] === undefined) {
+            incomingString = TokenParser.parseToken(token, '', incomingString);
+        }
+    });
+    return incomingString;
+};
+
+TokenParser.parseToken = function (Token, Value, Str) {
+    let regex = new RegExp("{(" + Token + ")}?", "g");
+    return Str.replace(regex, Value).replace('$', '');
```
Comment (seyilonge-nci) [inline src/utils/replaceTokens.js]: Can you replace the token "$" with "%" so as not to clash with js template literal

Comment (bryanpizzillo) [reply src/utils/replaceTokens.js]: So this one I don't like... `${FOO}` is a common placeholder string, so much so ES6 stole it. My next fallback is normal templating `{{FOO}}`.  `%{foo}` is odd -- if anything it should be `%FOO%` like DOS and other things that use % as vars. I think the linter in this case is dumb. 

So my preferences:
1. `${FOO}`
2. `{{FOO}}`
3. `%FOO%`

@arcepaul thoughts?

Comment (kate-mashkina) [reply src/utils/replaceTokens.js]: if turning off eslint warnings is not an option, then I'd prefer to use `{{FOO}}` format

Comment (bryanpizzillo) [reply src/utils/replaceTokens.js]: Didn't `{{FOO}}` also throw errors?

File: src/services/api/endpoints.js [hunk 2]
```diff
@@ -18,7 +18,8 @@ export const getEndpoint = ( endpoint ) => {
     // Define api endpoints here
     const urls = {
         expandChar: `/Terms/expand/${DICTIONARY_NAME}/${AUDIENCE}/${LANGUAGE}`,
-        termDefinition: `/Terms/${DICTIONARY_NAME}/${AUDIENCE}/${LANGUAGE}`
+        termDefinition: `/Terms/${DICTIONARY_NAME}/${AUDIENCE}/${LANGUAGE}`,
+        totalResults: `/Terms/count/${DICTIONARY_NAME}/${AUDIENCE}/${LANGUAGE}`
```
[the remark this answers is not in the document]
Comment (kate-mashkina) [reply src/services/api/endpoints.js]: renamed, thanks!

File: src/utils/replaceTokens.js (hunk 1 above)
Comment (kate-mashkina) [reply src/utils/replaceTokens.js]: Just pushed updated code - all checks are passed

Comment (seyilonge-nci) [reply src/utils/replaceTokens.js]: I would prefer `3`

Comment (bryanpizzillo) [reply src/utils/replaceTokens.js]: Do you have a reason for the preference @seyilonge-nci ? I mean, I added some words around my thinking.

Comment (seyilonge-nci) [reply src/utils/replaceTokens.js]: Just realized your preference was in order, and you had a fallback. I'm fine with normal templating as suggested.

Comment (arcepaul) [reply src/utils/replaceTokens.js]: I definitely don't like %{FOO} looking at it.  I like the handlebars style {{F00}} that Kate has moved to.

**Pull request merged**